lib/ES/BranchTracker.pm

```diff
@@ -75,13 +76,21 @@ sub delete_branch {
 sub write {
 #===================================
     my $self = shift;
-    my $new  = Dump( $self->shas );
+    my $to_save = dclone( $self->shas );
+    # Empty hashes are caused by new repos that are unused which shouldn't
+    # force a commit.
+    for my $repo ( keys %{ $to_save } ) {
+        unless ( keys %{ $to_save->{$repo} } ) {
+            delete $to_save->{$repo};
```

Comment:
My (admittedly ancient) belief is that you _really_ want to use `each` to express the "deleting from a hash I am currently iterating" pattern.

https://perldoc.perl.org/functions/each.html

The relevant bit:

> If you add or delete a hash's elements while iterating over it, the effect on the iterator is unspecified; for example, entries may be skipped or duplicated--so don't do that. Exception: It is always safe to delete the item most recently returned by each, so the following code works properly:

```perl
   while (my ($key, $value) = each %hash) {
        print $key, "\n";
        delete $hash{$key};   # This is safe
    }
```